pkg/webconfig/config_test.go

```diff
@@ -15,171 +15,171 @@
-						LocalObjectReference: v1.LocalObjectReference{
-							Name: "test-configmap",
-						},
-						Key: "tls.crt",
-					},
-				},
-				KeySecret: v1.SecretKeySelector{
-					LocalObjectReference: v1.LocalObjectReference{
-						Name: "test-secret",
-					},
-					Key: "tls.key",
-				},
-			},
-			expectedData: `tls_server_config:
-  cert_file: /web_certs_path_prefix/configmap_test-configmap_tls.crt
-  key_file: /web_certs_path_prefix/secret_test-secret_tls.key
-`,
-		},
-		{
-			name: "minimal TLS config with client CA from configmap",
-			webTLSConfig: &monitoringv1.WebTLSConfig{
-				Cert: monitoringv1.SecretOrConfigMap{
-					ConfigMap: &v1.ConfigMapKeySelector{
-						LocalObjectReference: v1.LocalObjectReference{
-							Name: "test-configmap",
-						},
-						Key: "tls.crt",
-					},
-				},
-				KeySecret: v1.SecretKeySelector{
-					LocalObjectReference: v1.LocalObjectReference{
-						Name: "test-secret",
-					},
-					Key: "tls.key",
-				},
-				ClientCA: monitoringv1.SecretOrConfigMap{
-					ConfigMap: &v1.ConfigMapKeySelector{
-						LocalObjectReference: v1.LocalObjectReference{
-							Name: "test-configmap",
-						},
-						Key: "tls.client_ca",
-					},
-				},
-			},
-			expectedData: `tls_server_config:
-  cert_file: /web_certs_path_prefix/configmap_test-configmap_tls.crt
-  key_file: /web_certs_path_prefix/secret_test-secret_tls.key
-  client_ca_file: /web_certs_path_prefix/configmap_test-configmap_tls.client_ca
-`,
-		},
-		{
-			name: "TLS config with all parameters from secrets",
-			webTLSConfig: &monitoringv1.WebTLSConfig{
-				ClientCA: monitoringv1.SecretOrConfigMap{
-					Secret: &v1.SecretKeySelector{
-						LocalObjectReference: v1.LocalObjectReference{
-							Name: "test-secret",
-						},
-						Key: "tls.ca",
-					},
-				},
-				Cert: monitoringv1.SecretOrConfigMap{
-					Secret: &v1.SecretKeySelector{
-						LocalObjectReference: v1.LocalObjectReference{
-							Name: "test-secret",
-						},
-						Key: "tls.crt",
-					},
-				},
-				KeySecret: v1.SecretKeySelector{
-					LocalObjectReference: v1.LocalObjectReference{
-						Name: "test-secret",
-					},
-					Key: "tls.keySecret",
-				},
-				ClientAuthType:           "RequireAnyClientCert",
-				MinVersion:               "TLS11",
-				MaxVersion:               "TLS13",
-				CipherSuites:             []string{"cipher-1", "cipher-2"},
-				PreferServerCipherSuites: &falseVal,
-				CurvePreferences:         []string{"curve-1", "curve-2"},
-			},
-			expectedData: `tls_server_config:
-  cert_file: /web_certs_path_prefix/secret_test-secret_tls.crt
-  key_file: /web_certs_path_prefix/secret_test-secret_tls.keySecret
-  client_auth_type: RequireAnyClientCert
-  client_ca_file: /web_certs_path_prefix/secret_test-secret_tls.ca
-  min_version: TLS11
-  max_version: TLS13
-  cipher_suites:
-  - cipher-1
-  - cipher-2
-  prefer_server_cipher_suites: false
-  curve_preferences:
-  - curve-1
-  - curve-2
-`,
-		},
-	}
+// func TestMakeSecret(t *testing.T) {
```

Comment:
Obviously we want to keep the unit test :)
You can use `k8s.io/client-go/kubernetes/fake` to get a fake Secret client (see [here](https://github.com/prometheus-operator/prometheus-operator/blob/17618aa437b6d284c7d5e310e1ab0eb67d6427c6/pkg/assets/store_test.go#L68) for an example).